website_sale_affiliate/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+import logging
+
+from odoo.http import request, route
+from odoo.addons.website_sale.controllers.main import WebsiteSale as Base
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(Base):
+    def _store_affiliate_info(self, **kwargs):
```

Comment:
in this case there's no real benefit and you are even writing more lines of code :stuck_out_tongue: 
Plus, as I understand the behavior you cannot have a `key` if no `id` is there so the last part of the code is totally useless if the 1st condition is not satisfied. That's why I wrapped the whole block inside ``if 'ref' in kwargs:``. I'm not a fun of too many IF blocks but in this case it makes it easier to read and understand what you are really doing :wink: 
Nevertheless, this is not blocking...